packages/@lwc/node-reactions/src/dom-patching/node.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { getOwnPropertyDescriptor, defineProperties } from '@lwc/shared';
+import { ReactionRecord, QualifyingReactionTypes } from '../types';
+import queueReactionsForSubtree from '../core/traverse';
+import { flushQueue } from '../core/reaction-queue';
+import { isQualifyingElement, isQualifyingHost, marker } from '../core/reactions';
+
+export default function() {
+    // caching few DOM APIs
+    const { appendChild, insertBefore, removeChild, replaceChild } = Node.prototype;
```

Comment:
It is no reused outside of the function body. So kept them as local variables instead of pulling them out.
Also, we need to get the patched version of the APIs, so dereferencing them just before patching them would guarantee that we will always get the patched versions of the APIs(synthetic-shadow).